backend/backend/api/routes.py

```diff
@@ -9,6 +9,9 @@
 from utils.logger import logger
 from .schemas import QueryRequestSchema, EventsRequestSchema, ResponseSchema
 
+import gradio as gr
```

Comment:
Importing gradio as a separate line can make the code harder to read. Consider importing it along with other imports.